invokeai/backend/prompting/conditioning.py

```diff
@@ -96,6 +97,7 @@ def get_uc_and_c_and_ec(
 
     c, options = compel.build_conditioning_tensor_for_prompt_object(positive_prompt)
     uc, _ = compel.build_conditioning_tensor_for_prompt_object(negative_prompt)
+    c, uc = compel.pad_conditioning_tensors_to_same_length(c, uc)
```

Comment:
Why are they padded to the same length? Should this have any effect on the output or do the beginning and end markers take care of that?

Reply:
because the negative and positive prompts are computed independently, we can end up in a situation where the positive prompt is eg 144 tokens but the negative is only 77 tokens. this causes issues down the line because of the way the positive and negative are batched, so here they're made to be the same length.

Reply:
Might be worth a comment? Your call.